core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -444,14 +444,20 @@ object Validator {
     }
   }
 
+  private def containsFragments(selectionSet: List[Selection]): Boolean =
+    selectionSet.exists {
+      case f: Selection.Field => containsFragments(f.selectionSet)
+      case _                  => true
```

Comment:
I think it would be nice to explicitly add the 2 cases here: first it's more readable and if some day we add a new case we'll have to update that.

Reply:
Funnily enough I had the same thought once I had a look at the PR again. Added 👍